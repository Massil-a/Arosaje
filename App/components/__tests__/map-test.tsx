import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import ActuMap from '../../app/actunav/actumap';
import * as Location from 'expo-location';
import AsyncStorage from '@react-native-async-storage/async-storage';

jest.mock('@react-native-async-storage/async-storage', () => ({
    setItem: jest.fn(),
    getItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
}));

// Mock du module expo-location
jest.mock('expo-location', () => ({
    requestForegroundPermissionsAsync: jest.fn().mockResolvedValue({ status: 'granted' }),
    getCurrentPositionAsync: jest.fn().mockResolvedValue({
        coords: {
            latitude: 37.78825,
            longitude: -122.4324,
        },
    }),
}));

describe('ActuMap', () => {
  it('renders correctly', async () => {
    const { toJSON } = render(
      <NavigationContainer>
        <ActuMap />
      </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
