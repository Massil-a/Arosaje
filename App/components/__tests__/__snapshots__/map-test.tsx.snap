// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActuMap renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        color="#668F80"
        size="large"
      />
    </View>
  </View>
</View>
`;
